ipaserver/plugins/certmap.py

```diff
@@ -91,6 +91,57 @@
 register = Registry()
 
 
+def check_maprule_is_for_trusted_domain(api_inst, options, entry_attrs):
+    """
+    Check that the certmap rule references altSecurityIdentities and
+    associateddomain in options is a trusted domain.
+
+    :param api_inst: API instance
+    :param options: options passed to the command
+                    at least ipacertmapmaprule and
+                    associateddomain options are expected for the check
+
+    :raises: ValidationError if altSecurityIdentities is present in the
+             rule but no Active Directory trusted domain listed in
+             associated domains
+    """
+    is_trusted_domain_required = False
+    # If no explicit option passed, fallback to the content of the entry
+    # This helps to catch cases when an associated domain value is removed
+    # while the rule requires its presence
+    maprule = options.get('ipacertmapmaprule',
+                          entry_attrs.get('ipacertmapmaprule'))
```

Comment:
should be entry_attrs.single_value.get('ipacertmaprule') otherwise a list is returned and the call to maprule.lower() fails.